powersimdata/network/usa_tamu/model.py

```diff
@@ -77,26 +77,22 @@ def _drop_interconnect(self):
 def check_and_format_interconnect(interconnect):
     """Checks interconnect.
 
-    :param str/list interconnect: interconnect name(s).
+    :param str/list/set interconnect: interconnect name(s).
     :return: (*list*) -- interconnect(s)
     :raises TypeError: if parameter has wrong type.
     :raises ValueError: if interconnect not found or combination of interconnect is not
         appropriate.
     """
     if isinstance(interconnect, str):
         interconnect = [interconnect]
-    if not isinstance(interconnect, list):
-        raise TypeError("interconnect must be a str or list of str")
+    if not isinstance(interconnect, (set, list)):
+        raise TypeError("interconnect must be one of str/list/set")
 
+    interconnect = list(set(interconnect))
```

Comment:
Since we are creating a list internally, should we replace `list` with `sorted` to achieve predictable results, no matter the returned order of set elements?